workers/unity/Packages/io.improbable.gdk.buildsystem/Tests/BuildContextTests.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Gdk.BuildSystem.Configuration;
+using NUnit.Framework;
+using UnityEditor;
+
+namespace Improbable.Gdk.BuildSystem.Tests
+{
+    [TestFixture]
+    public class BuildContextTests
+    {
+        [Test]
+        public void GetBuildContexts_for_unconfigured_worker_type_does_not_throw_for_local_build()
+        {
+            var buildContextFilter = BuildContextFilter.Local("DummyWorkerType");
+
+            List<BuildContext> buildContexts = null;
+            Assert.DoesNotThrow(() =>
+            {
+                buildContexts = BuildContext.GetBuildContexts(buildContextFilter);
+            });
+
+            Assert.AreEqual(0, buildContexts.Count);
+        }
+
+        [Test]
+        public void GetBuildContexts_for_unconfigured_worker_type_does_not_throw_for_cloud_build()
+        {
+            var buildContextFilter = BuildContextFilter.Cloud("DummyWorkerType");
+
+            List<BuildContext> buildContexts = null;
+            Assert.DoesNotThrow(() =>
+            {
+                buildContexts = BuildContext.GetBuildContexts(buildContextFilter);
+            });
+
+            Assert.AreEqual(0, buildContexts.Count);
+        }
+
+#if UNITY_EDITOR_WIN
+        [TestCase("UnityClient", 1, BuildTarget.StandaloneWindows64)]
+        [TestCase("UnityGameLogic", 1, BuildTarget.StandaloneWindows64)]
+        [TestCase("DummyWorkerType", 0, BuildTarget.StandaloneWindows64)]
+#elif UNITY_EDITOR_OSX
+        [TestCase("UnityClient", 1, BuildTarget.StandaloneOSX)]
+        [TestCase("UnityGameLogic", 1, BuildTarget.StandaloneOSX)]
+        [TestCase("DummyWorkerType", 0, BuildTarget.StandaloneOSX)]
+#endif
+        [TestCase("UnityClient", 0, BuildTarget.StandaloneLinux64)]
+        [TestCase("UnityGameLogic", 0, BuildTarget.StandaloneLinux64)]
+        [TestCase("DummyWorkerType", 0, BuildTarget.StandaloneLinux64)]
```

Comment:
If we inject the `BuildConfig` rather than using the project one we can avoid this

Reply:
not really, this is testing the filters that we can pass through cli

Reply:
I mean we don't need the ifdefs (since we can control the underlying `BuildConfig` object so its not OS dependent).